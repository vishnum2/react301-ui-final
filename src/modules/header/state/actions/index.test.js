import {login, logout, createUser} from './index';
import { LOGIN, LOGOUT } from "./types";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosWrapper from "../../../../apis/axiosCreate";

const mockStore = configureStore([thunk]);
const store = mockStore();

describe('test actions', () => {
    it('should return the proper response on login', () => {
        const object = {sample: 'test'};
        const response = login(object);
        expect(response).toEqual({
            type: LOGIN,
            payload: object
          });
    });

    it('should return proper response on logout', () => {
        const response = logout();
        expect(response).toEqual({
            type: LOGOUT,
            payload: null
          });
    });

    it('should call post and resolve', () => {
        store.dispatch(createUser({}));
        expect(store.getActions()).toMatchSnapshot();
        
    })

});