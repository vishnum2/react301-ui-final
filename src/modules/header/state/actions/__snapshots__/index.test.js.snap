// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test actions should call post and resolve 1`] = `Array []`;
